templates/node.ovc/actions.py

```diff
@@ -259,13 +263,21 @@ def install(job):
         if not machine:
             machine = _create_machine(service, space)
 
+        sshkey = service.producers['sshkey'][0]
+        key_path = j.sal.fs.joinPaths(sshkey.path, sshkey.name)
+
+        space.configure_machine(machine=machine, name=service.name, sshkey_name=sshkey.name, sshkey_path=key_path)
+
```

Comment:
why are you calling space.configure_machine even if the machine was newly created ?? the machine create will already call configure_machine.

Reply:
consider the following scenario
creating a machine --> our script will run  --> trying to get a machine with the name, which will not exist --> call _create_machine function --> the machine created --> configure_machine is being executed --> an error happened in the middle of configure_machine --> AYS will re-run --> it will try to get the machine --> the machine is already created --> will not get inside _create_machine function --> the missing part of the configuration of the machine will never be completed

Reply:
that is exactly what I am asking ?? you only need to call the configure function if the machine already exist.